ext/SymbolicsSymPyExt.jl

```diff
@@ -29,4 +29,174 @@ function Symbolics.symbolics_to_sympy(expr)
     end
 end
 
-end #module
+"""
+    sympy_to_symbolics(sympy_expr, vars)
+
+Converts a SymPy expression to a Symbolics.jl expression.
+
+This function takes a SymPy expression and a list or dictionary of Symbolics variables
+to ensure proper variable mapping during conversion. It is used to return solver results
+to the Symbolics.jl format.
+
+# Arguments
+- `sympy_expr`: The SymPy expression to convert (typically a `SymPy.Sym` object).
+- `vars`: A list or dictionary mapping variable names to Symbolics.jl variables.
+
+# Example
+```julia
+@variables x y
+sympy_expr = SymPy.Sym("x")^2 + SymPy.Sym("y")
+symbolics_expr = sympy_to_symbolics(sympy_expr, [x, y])
+```
+
+See also: [`symbolics_to_sympy`]
+"""
+function sympy_to_symbolics(sympy_expr, vars)
+    varmap = Dict(string(nameof(v)) => v for v in vars)
+    # Convert SymPy expression to string and parse back to Symbolics
+    expr_str = string(sympy_expr)
+    # Use Symbolics' parsing with variable mapping
+    return Symbolics.parse(expr_str, varmap)
+end
+
+# Wrapper Functions for SymPy Solvers
+
+"""
+    sympy_linear_solve(A, b)
+
+Solves a linear system Ax = b using SymPy's linear solver.
+
+# Arguments
+- `A`: A matrix of Symbolics.jl expressions.
+- `b`: A vector of Symbolics.jl expressions.
+
+# Returns
+A vector of Symbolics.jl expressions representing the solution.
+
+# Example
+```julia
+@variables x y
+A = [1 2; 3 4]
+b = [x, y]
+sol = sympy_linear_solve(A, b)
+```
+"""
+function sympy_linear_solve(A, b)
+    A_sympy = map(symbolics_to_sympy, A)
+    b_sympy = map(symbolics_to_sympy, b)
+    A_mat = SymPy.SymMatrix(A_sympy)
+    b_vec = SymPy.SymMatrix(b_sympy)
+    sol_sympy = A_mat \ b_vec  # SymPy's linear solver
+    vars = unique(vcat(vec(Symbolics.get_variables.(A)), Symbolics.get_variables.(b)))
+    return [sympy_to_symbolics(s, vars) for s in sol_sympy]
+end
+
+"""
+    sympy_algebraic_solve(expr, var)
+
+Solves an algebraic equation expr = 0 for the variable var using SymPy.
+
+# Arguments
+- `expr`: A Symbolics.jl expression representing the equation.
+- `var`: The Symbolics.jl variable to solve for.
+
+# Returns
+A list of Symbolics.jl expressions representing the solutions.
+
+# Example
+```julia
+@variables x
+expr = x^2 - 4
+sol = sympy_algebraic_solve(expr, x)
+```
+"""
+function sympy_algebraic_solve(expr, var)
+    expr_sympy = symbolics_to_sympy(expr)
+    var_sympy = symbolics_to_sympy(var)
+    sol_sympy = SymPy.solve(expr_sympy, var_sympy)
+    return [sympy_to_symbolics(s, [var]) for s in sol_sympy]
```

Comment:
This is only for scalar / single equations right? 

https://docs.sympy.org/latest/guides/solving/solve-system-of-equations-algebraically.html#solve-a-system-of-nonlinear-equations-algebraically